### PR TITLE
Collapse: updates component to use the new theme model

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR updates component to use the new theme model
**Which issue(s) this PR fixes**:

Relates to #33264

### CODE REVIEW

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ const getStyles = (theme: GrafanaTheme) => ({
       animation: loader 2s cubic-bezier(0.17, 0.67, 0.83, 0.67) 500ms;
       animation-iteration-count: 100;
       left: -25%;
-      background: ${theme.palette.blue85};
+      background: ${theme.v1.palette.blue85};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:47]: theme.palette.primary.main

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+const getStyles = (theme: GrafanaThemeV2) => ({
   collapse: css`
     label: collapse;
-    margin-bottom: ${theme.spacing.sm};
+    margin-bottom: ${theme.spacing(1)};
   `,
   collapseBody: css`
     label: collapse__body;
-    padding: ${theme.panelPadding}px;
+    padding: ${theme.v1.panelPadding}px;
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:15]: theme.spacing(theme.components.panel.padding)

**Review by torkelo — CHANGES_REQUESTED** [review]
looks good, just some minor things

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+const getStyles = (theme: GrafanaThemeV2) => ({
   collapse: css`
     label: collapse;
-    margin-bottom: ${theme.spacing.sm};
+    margin-bottom: ${theme.spacing(1)};
   `,
   collapseBody: css`
     label: collapse__body;
-    padding: ${theme.panelPadding}px;
+    padding: ${theme.spacing(theme.components.panel.padding)}px;
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:15]:
```suggestion
    padding: ${theme.spacing(theme.components.panel.padding)};
```

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 4]
```diff
@@ -29,7 +29,7 @@ const getStyles = (theme: GrafanaTheme) => ({
     position: relative;
     overflow: hidden;
     background: none;
-    margin: ${theme.spacing.xs};
+    margin: ${theme.spacing(4)};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:32]:
```suggestion
    margin: ${theme.spacing(0.5)};
```

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 5]
```diff
@@ -68,14 +68,14 @@ const getStyles = (theme: GrafanaTheme) => ({
...
     cursor: pointer;
-    padding: ${theme.spacing.sm} ${theme.spacing.md};
+    padding: ${theme.spacing(1, 2)};
   `,
   headerButtons: css`
     label: collapse__header-buttons;
-    margin-right: ${theme.spacing.sm};
-    margin-top: ${theme.spacing.xxs};
+    margin-right: ${theme.spacing(1)};
+    margin-top: ${theme.spacing(5)};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:76]:
```suggestion
    margin-top: ${theme.spacing(0.25)};
```

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 6]
```diff
@@ -68,14 +68,14 @@ const getStyles = (theme: GrafanaTheme) => ({
...
     font-size: ${theme.typography.size.lg};
-    line-height: ${theme.typography.heading.h6};
+    line-height: ${theme.typography.size.md};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:78]:
```suggestion
    line-height: ${theme.typography.h6.lineHeight};
```

**Review by torkelo — CHANGES_REQUESTED** [review]
found some mistakes

**Review by dprokop — APPROVED** [review]

**Review by dprokop — APPROVED** [review]

**Review by torkelo — APPROVED** [review]

**Pull request merged**